src/components/Recording.jsx

```diff
@@ -13,23 +16,38 @@ export default class Recording extends Component {
         this.onRecognitionEnd = this.onRecognitionEnd.bind(this);
         this.onRecognitionResult = this.onRecognitionResult.bind(this);
         this.upgradeRequired = this.upgradeRequired.bind(this);
+    }
 
+    componentDidMount() {
         if (!('webkitSpeechRecognition' in window)) {
             this.upgradeRequired();
         } else {
             this.initializeRecognition();
         }
+        this.props.resetSearch();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.results.length === 0 && nextProps.results.length > 0) {
+            browserHistory.push('/app/results');
+        }
+    }
+
+    componentWillUnmount() {
+        recognition.stop();
+        audioContext.close();
     }
 
     onRecognitionStart() {
         if (this.props.recording) {
-            console.log('stop recording');
+            // User pressing button to stop recording
             recognition.stop();
             return;
         }
 
-        console.log('start recording');
+        // User starting recording
         recognition.start();
+        // The code below is executed but immediately replaced if recognition starts successfully
         this.refs.micImg.src = '/static/img/mic-slash.gif';
```

Comment:
Actually, this is the only other reference to a gif file aside from the prefetch calls.